Gems/OpenXRVk/Code/Source/OpenXRVkInput.cpp

```diff
@@ -345,51 +371,78 @@ namespace OpenXRVk
 
-            LocateControllerSpace(device->GetPredictedDisplayTime(), session->GetXrSpace(OpenXRVk::SpaceType::View), static_cast<AZ::u32>(hand));
+            LocateControllerSpace(predictedDisplayTime, xrBaseSpaceForJoysticks, static_cast<AZ::u32>(hand));
         }
 
         // Cache 3d location information
         for (AZ::u32 i = 0; i < static_cast<AZ::u32>(SpaceType::Count); i++)
         {
             const auto spaceType = static_cast<SpaceType>(i);
-            LocateVisualizedSpace(device->GetPredictedDisplayTime(), session->GetXrSpace(spaceType),
-                                  session->GetXrSpace(OpenXRVk::SpaceType::View), spaceType);
+            LocateVisualizedSpace(predictedDisplayTime, session->GetXrSpace(spaceType),
+                xrBaseSpaceForVisualization, spaceType);
         }
 
-        // XR to AZ vector conversion...
-        // Goes from y-up to z-up configuration (keeping Right Handed system)
-        const auto convertVector3 = [](const XrVector3f& xrVec3) -> AZ::Vector3
-        {
-            return AZ::Vector3{ xrVec3.x, -xrVec3.z, xrVec3.y };
-        };
+        rawControllerData.m_leftPositionState = AzPositionFromXrPose(m_handSpaceLocation[static_cast<AZ::u32>(XR::Side::Left)].pose);
+        rawControllerData.m_rightPositionState = AzPositionFromXrPose(m_handSpaceLocation[static_cast<AZ::u32>(XR::Side::Right)].pose);
 
-        // XR to AZ quaternion conversion...
-        // Goes from y-up to z-up configuration (keeping Right Handed system)
-        const auto convertQuat = [](const XrQuaternionf& xrQuat) -> AZ::Quaternion
+        rawControllerData.m_leftOrientationState = AzQuaternionFromXrPose(m_handSpaceLocation[static_cast<AZ::u32>(XR::Side::Left)].pose);
+        rawControllerData.m_rightOrientationState = AzQuaternionFromXrPose(m_handSpaceLocation[static_cast<AZ::u32>(XR::Side::Right)].pose);
+
+        if (LocateEyeViews(predictedDisplayTime, xrViews))
         {
-            return AZ::Quaternion{ xrQuat.x, -xrQuat.z, xrQuat.y, xrQuat.w };
-        };
+            //! Time to notify the engine that we have new poses.
+            const auto& xrSpaceLocationHeadToBase = m_xrVisualizedSpaceLocations[OpenXRVk::SpaceType::View];
+            const auto baseToHeadTm = AzTransformFromXrPose(xrSpaceLocationHeadToBase.pose);
+            const auto headToLeftEyeTm = AzTransformFromXrPose(xrViews[0].pose);
+            const auto headToRightEyeTm = AzTransformFromXrPose(xrViews[1].pose);
+
+            AZ::RPI::XRSpaceNotificationBus::Broadcast(&AZ::RPI::XRSpaceNotifications::OnXRSpaceLocationsChanged,
+                baseToHeadTm, headToLeftEyeTm, headToRightEyeTm);
+        }
 
-        rawControllerData.m_leftPositionState = convertVector3(m_handSpaceLocation[static_cast<AZ::u32>(XR::Side::Left)].pose.position);
-        rawControllerData.m_rightPositionState = convertVector3(m_handSpaceLocation[static_cast<AZ::u32>(XR::Side::Right)].pose.position);
+        return false;
```

Comment:
It looks like every return path is false - what does the return value mean?

Reply:
Nice catch. Was missing a return true;. Fixed. Also  bumped the minor version of the OpenXRVK Gem. The new version is 1.0.1.